app/src/main/java/com/github/studydistractor/sdp/account/FirebaseCreateAccount.kt

```diff
@@ -0,0 +1,71 @@
+package com.github.studydistractor.sdp.account
+
+import android.util.Log
+import android.widget.Toast
+import androidx.compose.Context
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.FirebaseDatabase
+import javax.inject.Inject
+
+/**
+ * A class that implements the CreateAccInterface and provides methods to check if user input data
+ * is valid and to create an account in Firebase database.
+ */
+class FirebaseCreateAccount @Inject constructor(): CreateAccInterface{
+
+    private var auth = FirebaseAuth.getInstance()
+
+    private var db: FirebaseDatabase = FirebaseDatabase.getInstance()
+
+    /**
+     * Check if the given phone number is not empty and its length is exactly 10 characters
+     * @param phoneNumber The phone number to check
+     * @return true if the phone number is correctly formatted
+     */
+    @Inject
+    override fun checkPhoneFormat(phoneNumber: String): Boolean {
+        return !(phoneNumber.isEmpty() || phoneNumber.length != 10)
+    }
+
+    /**
+     * Check if the given name is not empty and it does not exceed or is equal to 255 characters
+     * @param name The name to check
+     * @return true if the name is correctly formatted
+     */
+    @Inject
+    override fun checkNameFormat(name: String): Boolean {
+        return !(name.isEmpty() || name.length >= 255)
+    }
+
+
+    /**
+    * Method to create a user account in Firebase database.
+    * @param year the user's year of birth
+    * @param month the user's month of birth
+    * @param day the user's day of birth
+    * @param firstname the user's first name
+    * @param lastname the user's last name
+    * @param phone the user's phone number
+    */
+    @Inject
+    override fun postData(year: Int, month: Int, day: Int, firstname: String, lastname: String, phone: String){
```

Comment:
Well, it was my first plan but after some discussion with Willy, we conclude that it was not useful. Maybe it would an interesting discussion to have together

Reply:
We definitely need to restructure our code and be **consistent**. So, if we plan to do this for any other piece of data (which is a good practice), we should do it as soon as possible.

IMO you should only resolve if you don't have time to do it.